rise/static/main.js

```diff
@@ -857,47 +867,56 @@ define([
    * returns null if no match is found
    */
   function reveal_cell_index(notebook, cell_type=null, auto_select_fragment=false) {
-    var [slide, subslide, fragment] = reveal_current_position();
-
-    /* just scan all cells until we find one at that address
-     * except that we need to start at -1 or 0 depending on
-     * whether the first slide has a slide tag or not
+    /* scan all cells until we find one that matches current reveal location
+     * need to deal carefully with beginning of that process because
+     * (.) we do not impose a starting 'slide', and
+     * (.) the first cell(s) might be of type 'skip'
+     *     which then must not be counted
      */
-    var slide_counter = is_slide(notebook.get_cell(0)) ? -1 : 0;
-    var subslide_counter = 0;
-    var fragment_counter = 0;
-    var result = null;
-
-    notebook.get_cells().forEach(function (cell, index) {
-      if (result) {
-        // keep it short: skip if we found already
-        return;
-      }
-      if (is_slide(cell)) {
-        slide_counter += 1;
-        subslide_counter = 0;
-      } else if (is_subslide(cell)) {
-        subslide_counter += 1;
-      }
+    let [slide, subslide, fragment] = reveal_current_position();
+
+    // start at slide -1 because we don't impose a starting 'slide'
```

Comment:
Yep, and that's important because I know a lot of people actually skipping their first cells...

Reply:
yes, the first slide may not be explicitly marked
but issue #449 was about a more specific corner case, with one first cell marked as skip, and then a slide cell; in this case the previous code was off by one slide

Reply:
Yep, gotcha! Thanks for taking care of the corner case.